packages/govuk-frontend/src/govuk/errors/index.mjs

```diff
@@ -0,0 +1,33 @@
+/**
+ * A base class for `Error`s thrown by GOV.UK Frontend.
+ *
+ * It is meant to be extended into specific types of errors
+ * to be thrown by our code.
+ *
+ * @example
+ * ```js
+ * class MissingRootError extends GOVUKFrontendError {
+ *   // Setting an explicit name is important as extending the class will not
+ *   // set a new `name` on the subclass. The `name` property is important
+ *   // to ensure intelligible error names even if the class name gets
+ *   // mangled by a minifier
+ *   name = "MissingRootError"
+ * }
+ * ```
+ * @abstract
```

Comment:
Worth enforcing this in the constructor too?

```js
export class GOVUKFrontendError extends Error {
  name = 'GOVUKFrontendError'

  constructor () {
    if (this.constructor == GOVUKFrontendError) {
      throw new Error("GOVUKFrontendError is an abstract class and can't be instantiated");
    }
  }
}
```

Or is errors throwing errors too confusing?

Reply:
I'm not sure it's worth enforcing the `@abstract` in code. I'd rather lean towards shipping less code and if we get evidence that people misuse it, clamp down on it rather than be pre-emptively defensive 😊